src/conf_mode/container.py

```diff
@@ -0,0 +1,177 @@
+#!/usr/bin/env python3
+#
+# Copyright (C) 2020 VyOS maintainers and contributors
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 2 or later as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import os
+
+from vyos.config import Config
+from vyos.configdict import dict_merge
+from vyos.configdict import node_changed
+from vyos.configdict import leaf_node_changed
+from vyos import ConfigError
+from vyos.util import call, cmd, process_named_running
+from vyos.template import render
+from vyos import airbag
+airbag.enable()
+
+
+def get_config(config=None):
+    if config:
+        conf = config
+    else:
+        conf = Config()
+    base = ['container']
+    container = conf.get_config_dict(base, key_mangling=('-', '_'), get_first_key=True)
+
+    if 'name' in container or 'network' in container:
+        container['configured'] = True
+
+    # Delete container network, delete containers
+    dict = {}
+    tmp_net = node_changed(conf, ['container', 'network'])
+    if tmp_net:
+        dict = dict_merge({'net_remove' : tmp_net}, dict)
+        container.update(dict)
+
+    tmp_name = node_changed(conf, ['container', 'name'])
+    if tmp_name:
+        dict = dict_merge({'container_remove' : tmp_name}, dict)
+        container.update(dict)
+
+    return container
+
+def verify(container):
+    # bail out early - looks like removal from running config
+    if not container:
+        return None
+    else:
+        # Start docker
+        docker_pid = process_named_running('dockerd')
+        if not docker_pid:
+            call('systemctl restart docker.service')
+
+    # Add new container
+    if 'name' in container:
+        for name in container['name']:
```

Comment:
please use `for container, container_config in container['name'].items()` which makes the later code easiert to read.

`container_config` holds the same data as `container['name'][name]`

Reply:
done